tests/psimulate/test_branches.py

```diff
@@ -52,3 +55,20 @@ def test_calculate_random_seeds_existing(seed_count):
     assert len(seeds) == len(set(seeds))
     assert len(seeds + existing) == len(set(seeds + existing))
     assert len(seeds) == seed_count
+
+
+def test_keyspace_order():
+    ## divide an integer range of 15 into five bins
+    input_draw, random_seed, foo, bar, baz = [list(range(i, i + 3)) for i in range(0, 15, 3)]
+    branches = [
+        {"foo": foo, "bar": bar, "baz": baz} for foo, bar, baz in product(foo, bar, baz)
+    ]
+    # Define the correct ordering of the keyspace
+    ordered_keys = product(input_draw, random_seed, branches)
```

Comment:
It's not clear to me what this ordering actually is. I believe you when you say it's what we want it to be, but is there a way to make this clearer?

Reply:
I removed 'bar' and 'baz' and substituted a list comprehension for `product`--then, we don't have write out the list in full, but the ordering is more explicit based on the order of iterables.